snapcraft_legacy/internal/project_loader/__init__.py

```diff
@@ -53,3 +62,38 @@ def replace_attr(
         return result
 
     return attr
+
+
+def _validate_replacement(attr: str, replacement: str, value: Optional[str]) -> None:
```

Comment:
The "replacement" variable name sounds a bit strange to me as it's the thing that will be replaced and not the new value. Maybe it could be better expressed as "variable", "name" or something similar?

Reply:
Thanks, I improved the naming!